qutebrowser/components/braveadblock.py

```diff
@@ -286,11 +281,19 @@ def on_method_changed() -> None:
 def init(context: apitypes.InitContext) -> None:
     """Initialize the Brave ad blocker."""
     global ad_blocker
+    global adblock
+    global _outdated_version
+
+    if adblock is not None:
+        _adblock_info = version.MODULE_INFO["adblock"]
+        if _adblock_info.is_outdated():
+            adblock = None  # type: ignore[assignment]
+            _outdated_version = _adblock_info.get_version()
 
     if adblock is None:
```

Comment:
Given that we only check `adblock is None` here, couldn't you do something like:

```python3
_adblock_info = version.MODULE_INFO["adblock"]

if adblock is None or _adblock_info.is_outdated():
    ad_blocker = None  # type: ignore[unreachable]
    _outdated_version = _adblock_info.get_version()
    _possibly_show_missing_dependency_warning()
    return
```

To deal with less globals here and simplify the conditional a bit?

Reply:
Yes. This works because `get_version()` returns `None` if the module is not installed.